module/VuFind/src/VuFind/Controller/RatingsController.php

```diff
@@ -0,0 +1,134 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.    See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Controller
+ * @author   Jaro Ravila <jaro.ravila@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Page
+ */
+
+namespace VuFind\Controller;
+
+use VuFind\Exception\Forbidden as ForbiddenException;
+use VuFind\Validator\CsrfInterface;
+
+/**
+ * Ratings controller.
+ *
+ * @category VuFind
+ * @package  Controller
+ * @author   Jaro Ravila <jaro.ravila@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Page
+ */
+class RatingsController extends AbstractBase
+{
+    /**
+     * Get all ratings for the logged in user
+     *
+     * @return View
+     */
+    public function userRatingsAction()
+    {
+        $user = $this->getUser();
+        if (!$user) {
+            return $this->forceLogin();
+        }
+        if (!$this->ratingsEnabled()) {
+            throw new ForbiddenException('Ratings disabled.');
+        }
+        $limit = $this->getService(\VuFind\Config\AccountCapabilities::class)->getUserContentPageSize();
+        $page = $this->params()->fromQuery('page', 1);
+        $sort = $this->params()->fromQuery('sort', 'created desc');
+        $service = $this->getDbService(\VuFind\Db\Service\RatingsServiceInterface::class);
+        $ratings = $service->getRatingsPaginator(
+            $user->getId(),
+            $limit,
+            $page,
+            $sort,
+        );
+        $sortList = [
+            'created desc'  => [
+                'desc' => 'sort_created_desc',
+                'url' => '?sort=' . urlencode('created desc'),
+                'selected' => $sort === 'created desc',
+            ],
+            'created asc'  => [
+                'desc' => 'sort_created_asc',
+                'url' => '?sort=' . urlencode('created asc'),
+                'selected' => $sort === 'created asc',
+            ],
+            'title'  => [
+                'desc' => 'sort_title',
+                'url' => '?sort=' . urlencode('title'),
+                'selected' => $sort === 'title',
+            ],
+        ];
+        $recordLoader = $this->serviceLocator->get(\VuFind\Record\Loader::class);
+        $ids = [];
+        foreach ($ratings as $rating) {
+            $ids[] = $rating['source'] . '|' . $rating['record_id'];
+        }
+        $records = $recordLoader->loadBatch($ids, true);
+        foreach ($ratings as $i => $c) {
+            $c['recordTitle'] = $records[$i]->getTitle() ?? '';
```

Comment:
Does this do anything? I thought `$c` would have to be a reference in order for changes to get applied to it. Maybe I'm misunderstanding/misremembering how arrays are handled in foreach loops, though!

Reply:
`$c` here is actually an ArrayObject, so it works as it is passed as a reference by default

Reply:
Now it is `&$c` which should keep this working with plain arrays